src/docfx/build/xref/ExternalXrefMapLoader.cs

```diff
@@ -48,26 +49,23 @@ public static ExternalXrefMap Load(Config config, FileResolver fileResolver, Err
                     }
                     else
                     {
-                        var externalXrefSpecAndExternalXref = LoadJsonFile(physicalPath);
+                        var (xrefSpecs, xrefs) = LoadJsonFile(physicalPath);
 
                         // Fast pass for JSON xref files
-                        foreach (var (uid, spec) in externalXrefSpecAndExternalXref.externalXrefSpec)
+                        foreach (var (uid, spec) in xrefSpecs)
                         {
                             // for same uid with multiple specs, we should respect the order of the list
                             externalXrefMap.TryAdd(uid, spec);
                         }
-                        externalXref = externalXrefSpecAndExternalXref.externalXref;
+                        externalXref.AddRange(xrefs);
                     }
-                }
-
-                externalXrefMap.TrimExcess();
-                externalXref.TrimExcess();
+                });
 
-                return new ExternalXrefMap(externalXrefMap, externalXref);
+                return new ExternalXrefMap(externalXrefMap.AsDictionary(), externalXref.AsList());
             }
         }
 
-        public static (List<(string, Lazy<ExternalXrefSpec>)> externalXrefSpec, List<ExternalXref> externalXref) LoadJsonFile(string filePath)
+        internal static (List<(string, Lazy<ExternalXrefSpec>)> externalXrefSpec, List<ExternalXref> externalXref) LoadJsonFile(string filePath)
```

Comment:
nit: the class is `internal` already

Reply:
Yes, all of our class are `internal` by default. I marked the method as `internal` to express that it is only consumed by unit tests and not supposed to be called by other classes.